### PR TITLE
Update Prysm to V0.10.1

### PR DESCRIPTION
**Pull request by nisdas**
This PR updates Prysm to v0.10.1 of the spec. 

- [x] Update the BLS API to conform with the specified draft from the [ietf](https://tools.ietf.org/html/draft-irtf-cfrg-bls-signature-00)
- [x] Add in Signing Root Container to use for signing objects. 
- [x] Add `ComputeSigningRoot` to take in the object and domain to return a valid root
for a private key to sign.
- [x] Fix references across the repo to remove usage of domain where required.
- [x] Check for clock disparity in gossip messages.
- [x] Fix spec tests
- [x] Regenerate Benchmark Files

### CODE REVIEW

File: beacon-chain/rpc/validator/exit_test.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/exit_test.go]:
```suggestion
```

File: beacon-chain/rpc/validator/server_test.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/server_test.go]:
```suggestion
```

File: shared/keystore/deposit_input_test.go [hunk 3]
```diff
@@ -4,9 +4,10 @@ import (
 	"bytes"
 	"testing"
 
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+
```
Comment (rauljordan) [inline shared/keystore/deposit_input_test.go]:
```suggestion
```

File: shared/testutil/deposits.go [hunk 4]
```diff
@@ -4,6 +4,8 @@ import (
 	"sync"
 	"testing"
 
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+
```
Comment (rauljordan) [inline shared/testutil/deposits.go]:
```suggestion
```

File: shared/testutil/helpers.go [hunk 5]
```diff
@@ -6,15 +6,15 @@ import (
 	"math/rand"
 	"testing"
 
+	"github.com/prysmaticlabs/prysm/shared/params"
+
```
Comment (rauljordan) [inline shared/testutil/helpers.go]:
```suggestion
```

File: validator/client/validator_attest.go [hunk 6]
```diff
@@ -6,11 +6,12 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+
```
Comment (rauljordan) [inline validator/client/validator_attest.go]:
```suggestion
```

File: validator/client/validator_propose.go [hunk 7]
```diff
@@ -6,11 +6,12 @@ import (
 	"encoding/binary"
 	"fmt"
 
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+
```
Comment (rauljordan) [inline validator/client/validator_propose.go]:
```suggestion
```

**Comment by prestonvanloon** [comment]
@nisdas looks like tests fail. Is this branch good to go?

**Comment by nisdas** [comment]
@prestonvanloon Yeap this branch should be good, seems like minor failures. Fixing them now

**Comment by nisdas** [comment]
Superseded by #5127

**Pull request closed without merging**